custom_components/kia_uvo/number.py

```diff
@@ -103,14 +113,22 @@ async def async_set_native_value(self, value: float) -> None:
             and self.vehicle.ev_charge_limits_dc == int(value)
         ):
             return
+        if (
+            self._description.key == V2L_DISCHARGE_LIMIT_KEY
+            and self.vehicle.ev_v2l_discharge_limit == int(value)
+        ):
+            return
 
         # set new limits
         if self._description.key == AC_CHARGING_LIMIT_KEY:
             ac = value
             dc = self.vehicle.ev_charge_limits_dc
-        else:
+        if self._description.key == DC_CHARGING_LIMIT_KEY:
             ac = self.vehicle.ev_charge_limits_ac
             dc = value
+        if self._description.key == V2L_DISCHARGE_LIMIT_KEY:
```

Comment:
If the above now filters to the key does this need to exist? Haven't looked closely at the code though.  

My impression is this if statement doesn't do anything.

Reply:
Code above guards setting the same value, in this place later we can add the code to set the value to the lower layer

Reply:
I think the code is correct and could be merged. 

I would prefere the use of elif which makes it more readable but in the code above it is also not used.